pkg/providers/instance/instance.go

```diff
@@ -380,19 +380,18 @@ func setVMPropertiesBillingProfile(vmProperties *armcompute.VirtualMachineProper
 
 // setNodePoolNameTag sets "karpenter.sh/nodepool" tag
 func setNodePoolNameTag(tags map[string]*string, nodeClaim *corev1beta1.NodeClaim) {
-	if val, ok := nodeClaim.Labels[corev1beta1.NodePoolLabelKey]; ok {
-		tags[NodePoolTagKey] = &val
-	}
+	nodePoolLabel := nodeClaim.Labels[corev1beta1.NodePoolLabelKey]
+	tags[NodePoolTagKey] = &nodePoolLabel
```

Comment:
Why did we change this?

Reply:
Oh, good catch. This is unrelated to the change.

Pretty sure its a bug I found months back. Will separate it out into a new PR.